Get user media constraints resizemode (#41153)
mdn/content: files/en-us/web/api/mediadevices/getusermedia/index.md

});
```

The browser will try to honor the constraints, and will return a matching track if supported by the underlying hardware.
If not supported, the browser may attempt to crop and downscale a higher resolution stream from the underlying hardware in order to match the constraint.

> This will commonly be the default behavior, but can be forced by setting the [`resizeMode`](/en-US/docs/Web/API/MediaTrackConstraints#resizemode) constraint to `crop-and-scale` (or disabled with `none`):

Reviewer: ```suggestion
This will commonly be the default behavior, but can be forced by setting the [`resizeMode`](/en-US/docs/Web/API/MediaTrackConstraints#resizemode) constraint to `crop-and-scale` (or disabled by setting it to `none`):
```

> If not supported, the browser may attempt to crop and downscale a higher resolution stream from the underlying hardware in order to match the constraint.
> This will commonly be the default behavior, but can be forced by setting the [`resizeMode`](/en-US/docs/Web/API/MediaTrackConstraints#resizemode) constraint to `crop-and-scale` (or disabled by setting it to `none`):

Author: FYI This is not _quite_ the "default" behaviour. If the resizeMode is not specified the browser will work choose behaviour based on a fitness score that takes into account specified constraints and _both_ a constraint of `none` and `crop-and-scale`. This modifies to remove the term default "it is still covered by "may attempt to".

```suggestion
If not supported, the browser may attempt to crop and downscale a higher resolution stream from the underlying hardware in order to match the constraint (and might also reduce the frame rate if that was constrained).
This behavior can be forced by setting the [`resizeMode`](/en-US/docs/Web/API/MediaTrackConstraints#resizemode) constraint to `crop-and-scale` (or disabled by setting it to `none`):
```

```js
getUserMedia({
  audio: true,
  video: { width: 1280, height: 720, resizeMode: "crop-and-scale" },
});
```

> The browser may return another resolutions if an exact match is not available and the source is not to be scaled.

Reviewer: ```suggestion
The browser may return another resolution if an exact match is not available and the source is not to be scaled.
```
